sycl/source/detail/program_impl.cpp

```diff
@@ -400,7 +405,8 @@ void program_impl::create_pi_program_with_kernel_name(
 template <>
 cl_uint program_impl::get_info<info::program::reference_count>() const {
   if (is_host()) {
-    throw invalid_object_error("This instance of program is a host instance");
+    throw invalid_object_error("This instance of program is a host instance",
+                               PI_INVALID_DEVICE);
```

Comment:
```suggestion
                               PI_INVALID_OPERATION);
```

Reply:
Whenever SYCL reports an `invalid_object_error`, I think it's more logically consistent to report a `PI_INVALID_DEVICE`/`CONTEXT`/`PROGRAM`/etc. where we can instead of `PI_INVALID_OPERATION` (unless we can get more descriptive with something like `CL_PROFILING_INFO_NOT_AVAILABLE`).

Reply:
My way to interpret is "I call a method, get an exception saying INVALID_OPERATION, go to spec and understand that I should not call such method of the host program."
While if I get INVALID_CONTEXT I can think that the context is broken for some reason, but in reality it's perfectly valid.

I don't know what is more logical for other people, so I'm OK with Sergey's suggestion.

Reply:
Personally, I do agree with that line of thinking, it's just that SYCL spec chooses to interpret these cases as "invalid objects", so I think our reported error codes should follow suit.

Reply:
Done.